src/unlinkcommand.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import Command from '../core/command/command.js';
+import Range from '../engine/model/range.js';
+
+/**
+ * The unlink command. It is used by the {@link Link.Link link feature}.
+ *
+ * @memberOf link
+ * @extends core.command.Command
+ */
+export default class LinkCommand extends Command {
+	/**
+	 * Executes the command.
+	 *
+	 * When selection is collapsed then whole link will be unlinked. {@link engine.model.TreeWalker TreeWalker} is
+	 * looking for link bounds going forward and backward from the caret position.
+	 *
+	 * If selection is non-collapsed then only selected range will be unlinked.
+	 *
+	 * @protected
+	 */
+	_doExecute() {
+		const document = this.editor.document;
+		const selection = document.selection;
+		const ranges = selection.getRanges();
+		let rangesToUpdate = [];
+
+		document.enqueueChanges( () => {
+			// When selection is collapsed we have to find link bounds.
+			if ( selection.isCollapsed ) {
+				// Loop through each selection.
+				for ( let range of ranges ) {
+					// Get searching area.
+					const parentNodeRange = Range.createIn( range.start.parent );
+
+					// Create walker instances for going forward and backward.
+					const walker = parentNodeRange.getWalker( { startPosition: range.start } );
```

Comment:
My first thought was to do that as you wrote, but then I thought "We have a TreeWalker, and I'm sure that someone will remind mi about it on review :)". I'll try to do it without `TreeWalker`.

> BTW. I haven't tried it, but isn't this implementation removing links in two separate blocks > (i.e. paragraphs) if they are same?

I have test for that case: https://github.com/ckeditor/ckeditor5-link/pull/7/files#diff-de987b63569829c1dcab5b439a1d4b2bR113